backend/src/test/java/gov/cdc/usds/simplereport/service/email/EmailServiceTest.java

```diff
@@ -0,0 +1,41 @@
+package gov.cdc.usds.simplereport.service.email;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import com.sendgrid.helpers.mail.Mail;
+import gov.cdc.usds.simplereport.service.BaseServiceTest;
+import gov.cdc.usds.simplereport.test_util.SliceTestConfiguration.WithSimpleReportSiteAdminUser;
+import java.io.IOException;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+
+class EmailServiceTest extends BaseServiceTest<EmailService> {
+  @MockBean EmailProviderWrapper mockSendGrid;
```

Comment:
This approach is fine, but could be made simpler: since the only dependencies of `EmailService` are the `EmailProvider` and some properties, we should use constructor injection in EmailService, and then you don't have to extend BaseServiceTest at all, you can just use vanilla JUnit:

```java
    EmailProviderWrapper provider = Mockito.mock();
    SendgridConfigurationProperties props = new SendgridConfigurationProperties(true, "fake-key"); // made this up obviously
    EmailService service = new EmailService(provider, props);
    service.send(...) // etc
```

The advantage being that since its a pretty pure unit test, you can run it without a Spring context, and it will be speedier and less prone to weird interactions with other tests.

Reply:
We tried this several different ways, but when mocking EmailProvider, instantiating a fresh SendgridConfigurationProperties object, and then newing up a fresh EmailService with those mocks, it throws NPEs that don't make any sense. Given that this test works as-is and the performance implications on our already-longish test run are marginal, I'd suggest that we leave this working as-is

Would love to pair with you @benwarfield-usds at some point though, so we could maybe get to the root cause of our confusion here